includes/geocoders/Maps_GoogleGeocoder.php

```diff
@@ -35,7 +35,15 @@ public static function register() {
 	 * @return string
 	 */	
 	protected function getRequestUrl( $address ) {
-		return 'http://maps.googleapis.com/maps/api/geocode/xml?address=' . urlencode( $address ) . '&sensor=false';
+		$urlArgs = [
+			'address' => urlencode( $address ),
+			'sensor' => false
+		];
+		if ( !empty( $GLOBALS['egMapsGMaps3ApiKey'] ) ) {
```

Comment:
I'll merge, but beware this it's better to do `$var !== ''`. The behaviour of `empty` is weird, and it makes the code less clear (as you can't immediately see the type of $var)